### PR TITLE
docs(listbox-button): add labelling examples #1346

### PR DESCRIPTION
**Pull request by ianmcburnie**
# Issue #1346

## Part 1 of 2

Adds an example of correct labelling of stateful menu buttons.

* updates `menu-button.js` with additional logic to handle this (copied over from MIND patterns example)
* adds example to single select menu button section

Other

* updated eslint config to `ecmaVersion": 2020` to account for [optional chaining](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Optional_chaining) syntax.
* added a quick fix in `menu.js` for IE11 (CustomEvent polyfill was missing)

### Screenshots

Screen Shot 2021-03-11 at 4 58 58 PM

## Part 2 of 2

Adds 3 different examples of labelling listbox button. This is mostly just markdown, and a small change to `listbox-button.js`

### Screenshots

Screen Shot 2021-03-11 at 9 36 51 PM

Screen Shot 2021-03-11 at 9 36 58 PM

Screen Shot 2021-03-11 at 9 37 04 PM

### CODE REVIEW

**Comment by austingardner** [comment]
Is there any reason why the listbox-button doesn't use some sort of `<ul> / <li> `?

**Review by austingardner — APPROVED** [review]

File: docs/src/js/base/menu-button.js [hunk 1]
```diff
@@ -20,8 +20,13 @@ function onMenuKeyDown(e) {
     }
 }
 
-function onMenuItemSelect() {
+function onMenuItemSelect(e) {
+    if (this._buttonPrefix && e.detail.el.dataset?.menuitemradioValue) {
+        this._buttonTextEl.innerText = this._buttonPrefix + e.detail.el.dataset?.menuitemradioValue;
```
Comment (austingardner) [inline docs/src/js/base/menu-button.js:25]: I'm glad we added optional chaining; I think it makes things easier for sure.

**Comment by ianmcburnie** [comment]
@austingardner We used to use list based markup as a sensible fallback in the past when listbox and option roles were not well supported by AT. Now that they are, I switched to div based markup to try and make it clearer that we have a listbox of options, not a list of list items. I could easily be persuaded to go back to ul/li though :-)

File: docs/src/js/base/menu-button.js (hunk 1 above)
Comment (ianmcburnie) [reply docs/src/js/base/menu-button.js:25]: Yes, I love it! Thank you for reminding me about it :-)

**Review by agliga — APPROVED** [review]

**Pull request merged**